NomicFoundation/hardhat-vscode
# chore: convert to monorepo

- Set up monorepo with following packages:
    - "client",
    - "server",
    - "coc",
    - "test/e2e",
    - "test/e2e/projects/main",
    - "test/protocol" 
- Remove `postinstall` script since the workspaces feature handles installing dependencies on packages
- Set `--ignore-platform` as default for install, since we need all the variants of `solidity-analyzer`
- Adjust `.vscodeignore` and bundling scripts accordingly with dependency hoisting

All tests pass, and from my manual testing it looks like the vsix is packed properly, and coc client/server are bundled correctly as well.

Note that when switching to this branch, or once this is merged, one should delete node modules and trigger a clean `yarn install` to avoid problems.

Closes #335
